src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2362,7 +2362,15 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                         Debug.Assert(reportedError);
                         return;
                     }
-
+                case BoundKind.UnconvertedCollectionLiteralExpression:
+                    {
+                        if (operand.Type is null)
```

Comment:
What is generally signaled by `operand.Type` being null? Lack of a best type among the collection elements?

Reply:
Yes.